### PR TITLE
feat(winget): support publishing additional locale manifests

### PR DESCRIPTION
**Pull request by MohammedAnasuddinZaid**
This adds support for publishing additional locale manifests alongside
the required default locale manifest in the winget pipe.

The change adds a new `additional_locales` config option to the winget
section. Each entry specifies a `locale` and optionally overrides any
localizable field (publisher, description, tags, release notes, etc).
Fields that are not set inherit the value from the default locale config.

WinGet uses the default locale manifest for the rich metadata and falls
back to it when a field is missing from an additional locale manifest.
This implementation follows that model by using `cmp.Or` to fall back
to the default locale values for any unset override field.

Validation rejects empty locales, locales that match the default locale,
and duplicate locales. Template errors in additional locale fields are
handled the same way as for the default locale.

Existing configurations without `additional_locales` are unaffected.
The default locale behavior is unchanged.

### CODE REVIEW

File: www/content/customization/publish/winget.md [hunk 1]
```diff
@@ -70,6 +70,30 @@ winget:
...
+    # The `locale` field is required and must not match the `default_locale`.
+    # Duplicate locales are rejected.
+    #
+    # Each additional locale manifest can override the following fields:
+    # publisher, publisher_url, publisher_support_url, privacy_url, author,
+    # package_name, homepage, license, license_url, copyright, copyright_url,
+    # short_description, description, tags, release_notes, release_notes_url,
+    # installation_notes.
+    #
+    # {{< g_inline_version "v2.19" >}}
```
Comment (caarlos0) [inline www/content/customization/publish/winget.md]:
```suggestion
    # {{< g_inline_version "v2.18-unreleased" >}}
```

**Review by caarlos0 — APPROVED** [review]

**Comment by caarlos0** [comment]
lgtm, one minor comment, + lint issues

thanks! and sorry for the delay reviewing this

File: internal/pipe/winget/winget.go [hunk 2]
```diff
@@ -258,7 +262,111 @@ func (p Pipe) doRun(ctx *context.Context, winget config.Winget, cl client.Releas
 		InstallationNotes:   winget.InstallationNotes,
 		ManifestType:        "defaultLocale",
 		ManifestVersion:     manifestVersion,
-	}, artifact.WingetDefaultLocale)
+	}, artifact.WingetDefaultLocale, winget.DefaultLocale); err != nil {
+		return err
+	}
+
+	return p.doAdditionalLocales(ctx, winget)
```
Comment (caarlos0) [inline internal/pipe/winget/winget.go:279]: Calling `doAdditionalLocales` only here means the version, installer, and default-locale artifacts have already been registered before any additional locale is validated. The loop can also register earlier locales before rejecting a later one. Because the `errAdditionalLocale*` values are `pipe.Skip`, that failure is treated as non-fatal and the release can publish a silently partial manifest set.

Please split validation from rendering: before the first `createYAML` in `doRun`, template and validate every `AdditionalLocales[i].Locale` (empty, equal to `DefaultLocale`, or duplicate) and return the prepared `winget` value. Only create artifacts after the entire preflight succeeds; `doAdditionalLocales` can then render the prepared entries. For example:

```go
winget, err = prepareAdditionalLocales(ctx, winget)
if err != nil {
	return err
}

// Create the version, installer, default locale, and additional locales only after preflight.
```

Please add a regression test with an invalid second locale and assert that no Winget artifacts were registered.

**Review by caarlos0 — COMMENTED** [review]
One correctness issue: an invalid additional locale can leave a partial set of Winget artifacts that still gets published. The inline comment includes a concrete preflight approach.

**Review by caarlos0 — COMMENTED** [review]
Re-reviewed at `8d697b0a7`; the previous issue remains unresolved. `doAdditionalLocales` is still called only after the version, installer, and default-locale artifacts are registered, and an earlier valid locale can also be registered before a later invalid locale returns `pipe.Skip`. Since that skip is non-fatal, a partial manifest set can still be published.

Please move all additional-locale identifier templating and validation into a preflight before the first `createYAML`, then render any artifacts only after the complete preflight succeeds. Also add a regression test with an invalid second locale that asserts zero Winget artifacts are registered.

**Comment by MohammedAnasuddinZaid** [comment]
@caarlos0 Thanks for the detailed review, and sorry for the late reply on this!

I've implemented all your feedback in commit `29f3084`:

1. **Preflight validation** — exactly as you suggested, I split validation from rendering. `prepareAdditionalLocales` now templates and validates every additional locale (empty, equal to `default_locale`, or duplicate) **before the first `createYAML`**, so an invalid locale can no longer leave a partially registered (and publishable) manifest set. `doAdditionalLocales` only renders the prepared entries after the preflight succeeds.

2. **Regression test** — added `TestRunNoArtifactsOnInvalidAdditionalLocale`: a valid first locale followed by an invalid second (duplicate) locale, asserting **zero** winget artifacts are registered.

3. **Lint issues** — fixed the var-block alignment, grouped the `const` declarations, and aligned the composite literal in `template.go` (gofumpt/golangci-lint clean).

4. **Docs badge** — corrected to `v2.18-unreleased`.

The branch is also rebased on the latest `main`. Could you take another look when you have a moment? The CI checks are waiting for your approval. Thanks again!

One more thing: GitHub is showing "Merging is blocked — Commits must have verified signatures." My commits aren't signed, so I can't merge even if everything is approved. Could you let me know if commit signing is required for fork PRs here, or if it can be waived in this case? If signing is required, I'll set up GPG/SSH signing and re-push. Thanks again!

**Comment by MohammedAnasuddinZaid** [comment]
@caarlos0 I pushed two more commits to fix the remaining failing checks:
1. test(winget) (93c7d51) — added a full-additional-locale case covering every override field, plus template-error cases for additional-locale fields. prepareAdditionalLocales is now at 100% coverage and the winget.go patch coverage is well above the target.
2. test(builders) (0bb05cb) — the windows-latest failures came from mlugg/setup-zig forcing a shared repo-root .zig-cache, which the rust and zig test packages corrupt when they build in parallel. I isolated the zig cache per test via ZIG_LOCAL_CACHE_DIR / ZIG_GLOBAL_CACHE_DIR.
The new workflow runs are awaiting maintainer approval — could you approve them and re-review when you have a moment?
Also, merging is still blocked on "Commits must have verified signatures." Is commit signing mandatory for fork PRs here, or can it be waived? If it's required, I'll set up signing and re-push the branch.

**Comment by MohammedAnasuddinZaid** [comment]
@caarlos0 hello — this PR (winget: additional locale manifests) now has everything green except two things that need your action:

1. The 8 workflow runs for the new head (0bb05cb) are awaiting your approval. The two new commits fix the flaky test (windows-latest) zig/rust failures — root cause was mlugg/setup-zig forcing a shared repo-root .zig-cache, which the rust and zig test packages corrupted while running in parallel. I've isolated the cache per test via ZIG_LOCAL_CACHE_DIR/ZIG_GLOBAL_CACHE_DIR, verified locally with go vet, golangci-lint (0 issues), and passing tests. Coverage is back above target (patch ~96%, prepareAdditionalLocales 100%, doAdditionalLocales 93.8%).

2. Could you merge with an admin override for the "verified signatures" requirement, or temporarily waive it? I'm a first-time contributor and haven't set up commit signing for this repo. If you'd rather not, I'm happy to set up GPG/SSH signing and re-sign all commits instead — just say the word.

Thanks!

**Comment by MohammedAnasuddinZaid** [comment]
@caarlos0 Quick final status on this PR:

Everything on my end is done — including the signature requirement. I set up GPG signing and re-signed all 4 commits (new head: d48b0a5). They now show as Verified on GitHub, so the "Commits must have verified signatures" block is resolved. Only the signatures (and author email) changed — the code content is byte-identical to what you already approved at 0bb05cb.

Current state:
- Review: approved by you
- All checks were green on the previous head; the force-push triggered a re-run
- The 8 workflows on the new head are awaiting your approval 
- The one neutral check (GitHub Advanced Security / grype — "1 configuration not found") is a benign Code Scanning quirk: 
scan-sboms only runs on main (cron/manual dispatch), so GitHub can't correlate alerts for a fork PR — it's not a failure and not blocking. The actual grype source scan passed.

What I need from you: approve the 8 pending workflow runs so the checks finish, and merge when green.

Thanks again for the thorough review and your patience on the slow turnarounds!

File: internal/pipe/winget/winget.go [hunk 3]
```diff
@@ -258,7 +272,128 @@ func (p Pipe) doRun(ctx *context.Context, winget config.Winget, cl client.Releas
...
+			ReleaseNotesURL:     cmp.Or(aloc.ReleaseNotesURL, winget.ReleaseNotesURL),
+			InstallationNotes:   cmp.Or(aloc.InstallationNotes, winget.InstallationNotes),
+			ManifestType:        "locale",
+			ManifestVersion:     manifestVersion,
+		}, artifact.WingetLocale, aloc.Locale); err != nil {
+			return err
+		}
+	}
+
+	return nil
```
Comment (caarlos0) [inline internal/pipe/winget/winget.go]: `range` already handles an empty slice, and the description normalization and release-note fallback do not need temporary branches. Iterating locale values is sufficient because this loop does not modify locale fields; `fixTags` keeps the same slice-aliasing behavior.

```suggestion
	for _, aloc := range winget.AdditionalLocales {
		tags := aloc.Tags
		if len(tags) == 0 {
			tags = winget.Tags
		}

		if err := createYAML(ctx, winget, Locale{
			PackageIdentifier:   winget.PackageIdentifier,
			PackageVersion:      ctx.Version,
			PackageLocale:       aloc.Locale,
			Publisher:           cmp.Or(aloc.Publisher, winget.Publisher),
			PublisherURL:        cmp.Or(aloc.PublisherURL, winget.PublisherURL),
			PublisherSupportURL: cmp.Or(aloc.PublisherSupportURL, winget.PublisherSupportURL),
			PrivacyURL:          cmp.Or(aloc.PrivacyURL, winget.PrivacyURL),
			Author:              cmp.Or(aloc.Author, winget.Author),
			PackageName:         cmp.Or(aloc.PackageName, winget.PackageName),
			PackageURL:          cmp.Or(aloc.Homepage, winget.Homepage),
			License:             cmp.Or(aloc.License, winget.License),
			LicenseURL:          cmp.Or(aloc.LicenseURL, winget.LicenseURL),
			Copyright:           cmp.Or(aloc.Copyright, winget.Copyright),
			CopyrightURL:        cmp.Or(aloc.CopyrightURL, winget.CopyrightURL),
			ShortDescription:    cmp.Or(aloc.ShortDescription, winget.ShortDescription),
			Description:         strings.ReplaceAll(aloc.Description, "\t", "  "),
			Moniker:             winget.Name,
			Tags:                fixTags(tags),
			ReleaseNotes:        cmp.Or(aloc.ReleaseNotes, winget.ReleaseNotes),
			ReleaseNotesURL:     cmp.Or(aloc.ReleaseNotesURL, winget.ReleaseNotesURL),
			InstallationNotes:   cmp.Or(aloc.InstallationNotes, winget.InstallationNotes),
			ManifestType:        "locale",
			ManifestVersion:     manifestVersion,
		}, artifact.WingetLocale, aloc.Locale); err != nil {
			return err
		}
	}

	return nil
```

**Review by caarlos0 — COMMENTED** [review]
One behavior-preserving simplification for the additional-locale renderer; targeted Winget tests pass.

**Comment by MohammedAnasuddinZaid** [comment]
@caarlos0 Thanks for the review — all feedback is now addressed on the new head 74733f07.

Applied your simplification in 298bd2f3 exactly as suggested in doAdditionalLocales:

- Dropped the redundant len(...) == 0 guard (range over an empty slice is a no-op)
- Value iteration (for _, aloc := range) since the loop never mutates locale fields
- Inlined Description normalization and ReleaseNotes fallback via cmp.Or
- Kept the tags fallback — fixTags preserves the same slice-aliasing behavior

The behavior is identical (verified via the existing golden tests for the empty/override cases).

Branch also updated — merged latest main (ee0e3c11, generated files only, zero conflicts) via signed merge 74733f07. No schema/config drift: I regenerated schema.json and it's byte-identical.

Local verification on the final head:

- go build ./..., go vet ./..., gofmt — all clean
- golangci-lint (repo config) — 0 issues
- go test ./internal/pipe/winget/... + internal/artifact + pkg/config — pass
- Regression test TestRunNoArtifactsOnInvalidAdditionalLocale asserts zero artifacts on an invalid second locale

The PR now shows up-to-date with main. Could you approve the 8 pending workflow runs (fork PR, awaiting maintainer approval) and re-approve when you have a moment? Since code changed after your last approval, I want the sign-off to reflect the final head.

**Comment by MohammedAnasuddinZaid** [comment]
@caarlos0 — this PR keeps getting dequeued by the merge queue, and I've confirmed it's not specific to this PR.

What happened:

- Added to merge queue → github-merge-queue bot removed it with: "removed this pull request from the merge queue due to no response for status checks."
- The identical sequence appears on #6750 and #6751: each was added to the queue twice and dequeued twice with that same reason, before being merged manually.

Root cause (repo-side, not PR-side):
The merge queue runs the required checks against a temporary gh-readonly-queue/* branch via the merge_group event. But currently only .github/workflows/build.yml has the merge_group trigger (added in ci: merge queue, 7a647a7). The other workflow-based checks never run there because their workflows only listen to push/pull_request:

- .github/workflows/lint.yml
- .github/workflows/codeql.yml
- .github/workflows/govulncheck.yml
- .github/workflows/gitleaks.yml
- .github/workflows/grype.yml
- .github/workflows/docs.yml
- .github/workflows/depsreview.yaml

Because those never execute on the merge_group branch, their statuses never report, GitHub's merge-queue wait times out, and the bot auto-dequeues the PR. This will affect every PR until either:

1. merge_group: is added to those workflows, or
2. the merge queue's required checks are limited to ones that actually run on merge_group.

This PR itself is ready:

- Approved, all checks green (19/19 successful, 1 neutral).
- Commits are GPG-signed and show as Verified on GitHub.
- No conflicts with the base branch — GitHub reports the changes can be cleanly merged.

Since #6750/#6751/#6752 were all merged manually after hitting the same dequeues, could you do the same for #6733 — or apply the merge_group fix so the queue works going forward? Happy to help with the workflow changes if useful. Thanks!

**Comment by MohammedAnasuddinZaid** [comment]
@caarlos0 — I just want to close this out properly 😄. This was my first contribution to a project of this scale, and honestly, this PR would be a fraction of what it became without your reviews.

Three rounds of feedback, and every single round made the code measurably better:

1. You caught the partial-publish correctness issue that I'd completely missed — that a non-fatal skip on a later invalid locale could leave a silently broken manifest set. The preflight refactor you suggested, plus the regression test asserting zero artifacts, is what turned this from "works" into "correct."
2. Your simplification of the renderer was genuinely cleaner — and the golden tests proved the behavior was byte-identical.
3. The little details you nudged — the version badge in the docs, lint alignment — taught me that the bar for merging isn't just "compiles and passes."

I also learned the hard way (via the merge queue) that commit signing, regenerated schema files, and a clean rebase are what separate a mergeable PR from a stuck one. All of that is now muscle memory.

And thank you for the patience with the CI — including letting me dig into the flaky windows-latest failures, which turned out to be a repo-wide zig cache issue worth fixing for everyone.

Your lgtm meant a lot. Looking forward to contributing more — this repo is a masterclass in how maintainers should mentor new contributors.

Thanks again, and congrats on the v3.0.0 milestone!
